users/views.py

```diff
@@ -440,21 +447,136 @@ def change_username_api_view(request: Request):
 
 
 @api_view(["PATCH"])
-def update_profile_api_view(request: Request):
-    user = request.user
+def update_profile_api_view(request: Request) -> Response:
+    user: User = request.user
     serializer = UserUpdateProfileSerializer(user, data=request.data, partial=True)
     serializer.is_valid(raise_exception=True)
 
+    assert isinstance(request.data, dict)
```

Comment:
you shouldn't have to assert this. Is it for type checking?
You can use `cast(dict, request.data)` (`from typing import cast`) to make type checking behave.
But a better solution is that at this point the serializer has validated data:
`data_to_check_for_spam = serializer.validated_data`

Reply:
That actually goes for everything below, you should be using `serializer.validated_data` instead of `request.data` all the time after `serializer.is_valid()` gets called.

Reply:
For context, this is because `is_valid` can sometimes mutate the request data, depending on the specific serializer, and stores it as `validated_data`. We have this behavior in the `ForecastWriteSerializer` for reference.

Reply:
+1 to this, we should use `serializer.validated_data` as much as possible